### PR TITLE
Update the dependencies to resolve vulnerabilities.

### PR DESCRIPTION
**Pull request by pallavit**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="Azure.Storage.Blobs" />
     <PackageReference Include="Microsoft.Azure.Amqp" />
     <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" />
+    <PackageReference Include="Newtonsoft.Json" VersionOverride="13.0.1"/>
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:21]: This doesn't seem correct. Azure.* libraries should not be depending on Newtonsoft.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj [hunk 2]
```diff
@@ -32,9 +32,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" />
     <PackageReference Include="Microsoft.Azure.Amqp" />
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" />
+    <PackageReference Include="Newtonsoft.Json" VersionOverride="13.0.1"/>
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: Why are we using a VersionOverride here? This is the same version listed in the packages.data.props.

Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: Nevermind, I see this is the Track 1 library.

File: sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj [hunk 3]
```diff
@@ -28,6 +28,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Amqp" />
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" />
+    <PackageReference Include="NETStandard.Library" VersionOverride="2.0.3"/>
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj:31]: Why is this necessary?

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:21]: Agree.  There's no reason that this package should be here.

File: sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj [hunk 4]
```diff
@@ -21,6 +21,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.KeyVault.Core" />
     <PackageReference Include="Microsoft.Azure.Storage.Blob" />
-    <PackageReference Include="Newtonsoft.Json" />
+    <PackageReference Include="NetStandard.Library" VersionOverride="2.0.3"/>
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj:24]: Why are we adding new dependencies?

File: sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj [hunk 5]
```diff
@@ -21,6 +21,7 @@
...
+    <PackageReference Include="Newtonsoft.Json" VersionOverride="13.0.1"/>
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj:25]: What's our test plan for this, and why make it local rather than at the root level?

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 2 above)
Comment (jsquire) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: I'd still ask the same question.  Why override locally rather than at the root level?

**Review by jsquire — CHANGES_REQUESTED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj (hunk 1 above)
Comment (pallavit) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:21]: This package is here purely because the runtime resolution ends up picking up an older version of the library so this is forcing the correct version resolution. Let me know if there is another way to do it.

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:21]: There is no use of anything JSON in this package.   There's nothing that should be picking up any version of Newtonsoft.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 2 above)
Comment (pallavit) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: I did this for the same reason as above. The aim is not to upgrade dependency of every package. Note, w/o the assembly updates runtime resolution was still picking up 13.0.1 and w/o including this we were downgrading our dependencies to 10 version.

My thought here was lets only upgrade the minimum necessary for 2 reasons
1. Restrict the area of change.
2. Given this is T1 and will be obsolete soon, I don't see a value in doing it across the board for t1 ecosystem.

File: sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj (hunk 3 above)
Comment (pallavit) [reply sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj:31]: Same as above. Assembly resolution for this with current dependency set was coming out to an older version causing build errors and wanting us to explicitly set the assembly version here.

File: sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj (hunk 4 above)
Comment (pallavit) [reply sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj:24]: They aren't new dependencies just dependency closures.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 2 above)
Comment (jsquire) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: But this dependency with a vulnerable version already exists in the packages file:  
https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L31

I don't understand the advantage to doing this locally.

File: sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj (hunk 3 above)
Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.EventHubs/src/Microsoft.Azure.EventHubs.csproj:31]: This also already exists in the packages file for T1 dependencies. I don't understand the value in overriding locally:
https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L341

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 2 above)
Comment (jsquire) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: Why not just bump all the T1 packages to a safe version?  Why let a vulnerable version still exist as the baseline?

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj (hunk 1 above)
Comment (pallavit) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:21]: I was purely going by the closure\build errors\warning to guide me but let me look closure. But I agree we should not need the override for sure here.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 2 above)
Comment (pallavit) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj:37]: It is not the newtonsoft that is vulnerable. If it were I would absolutely bump that too.

File: eng/Packages.Data.props [hunk 6]
```diff
@@ -120,6 +120,7 @@
     <PackageReference Update="Azure.Monitor.OpenTelemetry.Exporter" Version="1.3.0-beta.1" />
     <PackageReference Update="Azure.Monitor.OpenTelemetry.LiveMetrics" Version="1.0.0-beta.1" />
     <PackageReference Update="Azure.Monitor.Query" Version="1.1.0" />
+    <PackageReference Update="Azure.MixedReality.Authentication" version= "1.2.0" />
```
Comment (jsquire) [inline eng/Packages.Data.props:123]: nit:  This should come before the Monitor packages to keep them alphabetized.

**Comment by pallavit** [comment]
@JoshLove-msft , @jsquire please take another pass. I believe I fixed all the concerns you had.

**Review by jsquire — APPROVED** [review]

**Comment by jsquire** [comment]
> @JoshLove-msft , @jsquire please take another pass. I believe I fixed all the concerns you had.

Other than the nit about alphabetical order, looks good.

**Comment by pallavit** [comment]
@JoshLove-msft please let me know if you have any concerns before I merge.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft please let me know if you have any concerns before I merge.

LGTM!
